File: Sources/RoyalVNCKit/SDK/Connection/VNCConnectionDelegate.swift
#if canImport(FoundationEssentials)
import FoundationEssentials
#else
import Foundation
#endif

#if canImport(ObjectiveC)
@objc(VNCConnectionDelegate)
#endif
public protocol VNCConnectionDelegate: AnyObject {
#if canImport(ObjectiveC)
    @objc
#endif
	func connection(_ connection: VNCConnection,
					stateDidChange connectionState: VNCConnection.ConnectionState)
	
#if canImport(ObjectiveC)
    @objc
#endif
	func connection(_ connection: VNCConnection,
					credentialFor authenticationType: VNCAuthenticationType,
					completion: @escaping (_ credential: VNCCredential?) -> Void)
	
#if canImport(ObjectiveC)
    @objc
#endif
	func connection(_ connection: VNCConnection,
					didCreateFramebuffer framebuffer: VNCFramebuffer)
	
#if canImport(ObjectiveC)
    @objc
#endif
	func connection(_ connection: VNCConnection,
					didResizeFramebuffer framebuffer: VNCFramebuffer)

#if canImport(ObjectiveC)
    @objc
#endif
    func connection(_ connection: VNCConnection,
                    didUpdateFramebuffer framebuffer: VNCFramebuffer,
                    x: UInt16, y: UInt16,
                    width: UInt16, height: UInt16)

#if canImport(ObjectiveC)
    @objc
#endif
	func connection(_ connection: VNCConnection,
					didUpdateCursor cursor: VNCCursor)
}

File: Bindings/kotlin/RoyalVNCAndroidTest/royalvnc/src/main/java/com/royalapps/royalvnc/VncConnectionDelegate.kt
package com.royalapps.royalvnc

interface VncConnectionDelegate {
    fun connectionStateDidChange(
        connection: VncConnection,
        connectionState: VncConnectionState
    )

    fun authenticate(
        connection: VncConnection,
        authenticationRequest: VncAuthenticationRequest
    )

    fun didCreateFramebuffer(
        connection: VncConnection,
        framebuffer: VncFramebuffer
    )

    fun didResizeFramebuffer(
        connection: VncConnection,
        framebuffer: VncFramebuffer
    )

    fun didUpdateFramebuffer(
        connection: VncConnection,
        framebuffer: VncFramebuffer,
        x: Short,
        y: Short,
        width: Short,
        height: Short
    )

    fun didUpdateCursor(
        connection: VncConnection,
        cursor: VncCursor
    )
}